src/OrchardCore.Modules/OrchardCore.Sitemaps/Routing/SitemapRouteConstraint.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Sitemaps.Services;
+
+namespace OrchardCore.Sitemaps.Routing
+{
+    public class SitemapRouteConstraint : IRouteConstraint
+    {
+        public const string RouteKey = "sitemap";
+
+        public bool Match(HttpContext httpContext, IRouter route, string routeKey, RouteValueDictionary values, RouteDirection routeDirection)
+        {
```

Comment:
and / or maybe a route transformer as used for autoroute entries, will see.

That said we already have access to all autoroutes and home route but i assume you don't want to expose all of them. I will need a demo ;) Or just try it when it will be ready to run so that i could see how it works and how it is configured.